### PR TITLE
build: add husky to lint code & cmt msg 

### PR DESCRIPTION
**Pull request by ilhan007**
Use husky to automatically lint code and commit messages locally upon git push (pre-push hook).

### CODE REVIEW

File: .github/workflows/lint.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+name: Lint
```
Comment (pskelin) [inline .github/workflows/lint.yaml:1]: Is this file needed? If yes, maybe put a comment why, it doesn't seem related to the hooks.

File: package.json [hunk 2]
```diff
@@ -65,14 +65,20 @@
     "unlink-all": "wsrun unlink",
     "hash": "wsrun --exclude-missing hash",
     "copy-css": "copy-and-watch \"packages/base/dist/css/**/*\" packages/main/dist/resources/css/base/ && copy-and-watch \"packages/theming/dist/css/**/*\" packages/main/dist/resources/css/theming/ && copy-and-watch \"packages/main/dist/css/**/*\" packages/main/dist/resources/css/main/ && copy-and-watch \"packages/base/dist/css/**/*\" packages/fiori/dist/resources/css/base/ && copy-and-watch \"packages/theming/dist/css/**/*\" packages/fiori/dist/resources/css/theming/ && copy-and-watch \"packages/main/dist/css/**/*\" packages/fiori/dist/resources/css/main/",
-    "prerelease": "node ./.github/actions/pre-release.js"
+    "prerelease": "node ./.github/actions/pre-release.js",
+    "prepare": "husky install",
+    "husky:commit-msg": "commitlint -e",
+    "husky:pre-commit": "yarn lint"
```
Comment (pskelin) [inline package.json:71]: Just an idea, isn't pre-push a better place, in case you want to quickly submit some dirty code to switch branches? After all, we want to avoid the central build from failing and that will only happen after a push.

File: .github/workflows/lint.yaml (hunk 1 above)
Comment (ilhan007) [reply .github/workflows/lint.yaml]: I used to have it in the commit message description: "add lint github actions check", but not it's not related to the hooks, perhaps won't be needed any more, so I will remove it.

File: package.json (hunk 2 above)
Comment (ilhan007) [reply package.json]: pre-push is better

**Review by pskelin — APPROVED** [review]

**Review by fifoosid — APPROVED** [review]
